File: Algorithms/SelectionSort/python/selectionsort.py
# Python program for implementation of Selection Sort
 
import sys 
n = input("Enter elements:")
A=n.split()
# Traverse through all array elements 
for i in range(len(A)): 
	
	# Find the minimum element in remaining unsorted array 
	min_idx = i 
	for j in range(i+1, len(A)): 
		if A[min_idx] > A[j]: 
			min_idx = j 
			
	# Swap the found minimum element with the first element	
	if min_idx != i:
	    A[i], A[min_idx] = A[min_idx], A[i] 

# Driver code to test above 
print ("Sorted array:") 
for i in range(len(A)): 
	print(A[i]) 

File: Algorithms/SelectionSort/rust/SelectionSort.rs
fn selection_sort(array: &mut [i32]) {
    for i in 0..array.len() {
        let min_idx = array[i..].iter()
            .enumerate()
            .min_by_key(|&(_, v)| v)
            .map(|(i, _)| i)
            .unwrap_or(0);

        array.swap(i, min_idx + i);
    }
}


fn main() {

    let mut array = [ 5, 9, 7, 3, 1, 2, 6, 10, 8, 4];

    selection_sort(&mut array);
    println!("Sorted Array {:?}", array);
}

